src/system.rs

```diff
@@ -164,21 +210,28 @@ pub type TypeId = usize;
 /// Defines a cache that can be used to store resolved [Type]s of a [Schema]
 #[derive(Debug, Clone)]
 pub struct TypeStore {
+    imports: HashMap<String, TypeId>, // stores all the imported types of a schema
     ids_by_name: HashMap<String, TypeId>,
```

Comment:
Now that this _doesn't_ include the imported types, I suggest adding a comment to explain what it does store.